crates/oxc_linter/src/utils/jest/parse_jest_fn_new.rs

```diff
@@ -0,0 +1,560 @@
+use std::{borrow::Cow, cmp::Ordering};
+
+use oxc_ast::{
+    ast::{
+        Argument, CallExpression, Expression, IdentifierName, IdentifierReference, MemberExpression,
+    },
+    AstKind,
+};
+use oxc_semantic::AstNode;
+use oxc_span::{Atom, Span};
+
+use crate::context::LintContext;
+
+use crate::utils::jest::{is_pure_string, JestFnKind, JestGeneralFnKind, PossibleJestNode};
+
+pub fn parse_jest_fn_call<'a>(
+    call_expr: &'a CallExpression<'a>,
+    possible_jest_node: &PossibleJestNode<'a, '_>,
+    ctx: &LintContext<'a>,
+) -> Option<ParsedJestFnCall<'a>> {
+    let original = possible_jest_node.original;
+    let node = possible_jest_node.node;
+    let callee = &call_expr.callee;
+    // If bailed out, we're not jest function
+    let resolved = resolve_to_jest_fn(call_expr, original)?;
```

Comment:
`parse_jest_fn_call` is imported as `parse_jest_fn_call_new` for every rule now, and I only changed this line.